src/screen/MyIdentityScreen/index.tsx

```diff
@@ -1,42 +1,38 @@
-import React, {useContext, useMemo} from 'react';
 import {RouteProp, useRoute} from '@react-navigation/native';
-import {NetworkContext} from 'context/NetworkContext';
-import {ChainApiContext} from 'context/ChainApiContext';
-import useAccountDetail from 'src/api/hooks/useAccountDetail';
-import SetInfo from './SetInfo';
-import RequestJudgement from './RequestJudgement';
-import DisplayJudgement from './DisplayJudgement';
 import LoadingView from 'presentational/LoadingView';
 import SafeView, {noTopEdges} from 'presentational/SafeView';
+import React, {useMemo} from 'react';
+import {useAccountIdentityInfo} from 'src/api/hooks/useAccountIdentityInfo';
 import {DashboardStackParamList} from 'src/navigation/navigation';
 import {myIdentityScreen} from 'src/navigation/routeKeys';
+import DisplayJudgement from './DisplayJudgement';
+import RequestJudgement from './RequestJudgement';
+import SetInfo from './SetInfo';
 
 function MyIdentity() {
   const {
     params: {address},
   } = useRoute<RouteProp<DashboardStackParamList, typeof myIdentityScreen>>();
-  const {currentNetwork} = useContext(NetworkContext);
-  const {api} = useContext(ChainApiContext);
-  const {inProgress, display, isNaked, detail} = useAccountDetail(currentNetwork?.key || 'polkadot', address, api);
+  const {data, isLoading} = useAccountIdentityInfo(address);
 
   const content = useMemo(() => {
-    if (inProgress || !address) {
+    if (isLoading || !address) {
       return <LoadingView />;
     }
 
-    if (detail?.data?.judgements.length) {
+    if (data?.registration?.judgements.length) {
       // there is already judgements to display
```

Comment:
We could get info like this from `useAccountIdentityInfo` so it's explicit and we don't have to add comments.
e.g: `hasJudgements`, `hasIdentity`

Reply:
you mean something like 694b50a ?